libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -9,34 +9,48 @@ import {
 } from '@tmo/books/data-access';
 import { FormBuilder } from '@angular/forms';
 import { Book } from '@tmo/shared/models';
-import { Observable } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+import { Subject, Observable } from 'rxjs';
 
 @Component({
   selector: 'tmo-book-search',
   templateUrl: './book-search.component.html',
-  styleUrls: ['./book-search.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  styleUrls: ['./book-search.component.scss']
 })
-export class BookSearchComponent implements OnInit {
-  books$: Observable<ReadingListBook[]>;
+export class BookSearchComponent implements OnInit, OnDestroy {
+  books$: Observable<ReadingListBook[]> = this.store.select(getAllBooks);
+  private destroyedFormValues$: Subject<void> = new Subject();
   searchForm = this.fb.group({
     term: ''
   });
 
   constructor(
     private readonly store: Store,
     private readonly fb: FormBuilder
-  ) {}
+  ) { }
 
   get searchTerm(): string {
     return this.searchForm.value.term;
   }
 
   ngOnInit(): void {
-    // Using async pipe operator in html instead of subscribing in ts file
-    // No need to unsubscribe manually since we are using async in html
-    // Added change detection to onpush for faster page load
-    this.books$ = this.store.select(getAllBooks);
+    this.searchForm.get('term').valueChanges
+      .pipe(
+        debounceTime(500),
+        distinctUntilChanged(),
+        takeUntil(this.destroyedFormValues$))
+      .subscribe((val) => {
```

Comment:
use better param name other than val (good code is readable code :) )

Reply:
Done. I will take care of this in future. Thanks